reactive/kotlinx-coroutines-reactive/test/IntegrationTest.kt

```diff
@@ -148,35 +148,41 @@ class IntegrationTest(
                     }
                 })
             }
-        suspend fun<T> assertDetectsBadPublisher(operation: suspend Publisher<T>.() -> T,
-                                                 block: Subscriber<in T>.(n: Long) -> Unit) =
-            assertFailsWith<IllegalStateException> { publisher(block).operation() }
+        suspend fun <T> assertDetectsBadPublisher(
+            operation: suspend Publisher<T>.() -> T,
+            message: String? = null,
+            block: Subscriber<in T>.(n: Long) -> Unit,
+        ) =
+            assertFailsWith<IllegalStateException> { publisher(block).operation() }.let {
+                message == null || it.message == message
```

Comment:
This line doesn't actually test exception messages, the resulting boolean is ignored

Reply:
Yeah, sorry. Fixed in the commit that changes the whole error-passing mechanism here.